libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyParserException.java

```diff
@@ -86,7 +86,51 @@ public static PolicyParserException newPolicyParserException(
         }
     }
 
+    public static PolicyParserException newPolicyParserException(
+        XContentLocation location,
+        String policyName,
+        String scopeName,
+        String entitlementType,
+        PolicyIllegalArgumentException cause
+    ) {
+        if (scopeName == null) {
```

Comment:
when would scope name be null? can we assert it isn't null instead?

Reply:
Fixed.